asset-transfer-basic/README.md

```diff
@@ -0,0 +1,142 @@
+   ******** FAILED to run the application: Error: Identity not found in wallet: appUser
+```
+Delete the `/fabric-samples/asset-transfer-basic/application-javascript/wallet` directory
+and retry this application.
+
+The certificate authority must have been restarted and the saved certificates for the
+admin and application user are not valid. Deleting the wallet store will force these to be reset
+with the new certificate authority.
+
+To see the SDK workings, try setting the logging to show on the console before running
+```
+export HFC_LOGGING='{"debug":"console"}'
+```
+or log to a file
+```
+export HFC_LOGGING='{"debug":"./debug.log"}'
+```
+
+## Clean up
+When you are finished, you can bring down the test network.
+The command will remove all the nodes of the test network, and delete any
+ledger data that you created:
+
+```
+./network.sh down
+```
+Be sure to delete the wallet directory create by the application.
+The stored identities will no longer be valid when restarting the network.
+
+## Configuring and running a Hardware Security Module
+
+The javascript sample application for asset transfer basic includes
+support for a HSM. Below are the steps required to run a Hardware Security Module (HSM)
+simulator locally.
+
+**_Note:_**
+	When using the HSM simulator you will only be able to run the application once
+	without restarting the Fabric network and removing the wallet store.
+	Rerunning the application will restart the HSM simulator. The user's identity
+	information that has been saved to the wallet store will not exist in the
+	restarted HSM simulator.
+
+### Install SoftHSM
+
+In order to run the javascript application in the absence of a real HSM, a software
+emulator of the PKCS#11 interface is required.
+For more information please refer to [SoftHSM](https://www.opendnssec.org/softhsm/).
+
+SoftHSM can either be installed using the package manager for your host system:
+
+* Ubuntu: `apt-get install softhsm2`
+* macOS: `brew install softhsm`
+* Windows: **unsupported**
+
+Or compiled and installed from source:
+
+1. install openssl 1.0.0+ or botan 1.10.0+
+2. download the source code from <https://dist.opendnssec.org/source/softhsm-2.2.0.tar.gz>
+3. `tar -xvf softhsm-2.2.0.tar.gz`
+4. `cd softhsm-2.2.0`
+5. `./configure --disable-gost` (would require additional libraries, turn it off unless you need 'gost' algorithm support for the Russian market)
+6. `make`
+7. `sudo make install`
+
+### Specify the SoftHSM configuration file
+
+The configuration file required for the simulator has been created and is
+located in the test network directory. Set the following environment
+variable to point to a config file before running the javascript application.
+When the javascript application sees the following environment variable, it will
+attempt to connect to the HSM simulator.
+
+```bash
+export SOFTHSM2_CONF="../test-network/hsm/softhsm2.conf"
+```
+
+### Create a token to store keys in the HSM
+
```

Comment:
We may need more instructions on how to use softHSM. For instance, I issued the following command: 
```
Available slots:
Slot 742340698
    Slot info:
        Description:      SoftHSM slot ID 0x2c3f385a                                      
        Manufacturer ID:  SoftHSM project                 
        Hardware version: 2.6
        Firmware version: 2.6
        Token present:    yes
    Token info:
        Manufacturer ID:  SoftHSM project                 
        Model:            SoftHSM v2      
        Hardware version: 2.6
        Firmware version: 2.6
        Serial number:    5e1731c1ac3f385a
        Initialized:      yes
        User PIN init.:   yes
        Label:            ForFabric                       
Slot 1
    Slot info:
        Description:      SoftHSM slot ID 0x1                                             
        Manufacturer ID:  SoftHSM project                 
        Hardware version: 2.6
        Firmware version: 2.6
        Token present:    yes
    Token info:
        Manufacturer ID:  SoftHSM project                 
        Model:            SoftHSM v2      
        Hardware version: 2.6
        Firmware version: 2.6
        Serial number:                    
        Initialized:      no
        User PIN init.:   no
        Label:                                  
```

but was still able to use slot 0